### PR TITLE
Update RPC example to use new features

### PR DESCRIPTION
**Pull request by alexcamposruiz**
### Summary

Update RPC example to use new features.

### Details and comments

- Service discovery
- Built-in RPC QoS profile

### CODE REVIEW

File: examples/connext_dds/remote_procedure_call/c++11/USER_QOS_PROFILES.xml [hunk 1]
```diff
@@ -4,147 +4,33 @@
...
-        
-            <!-- QoS for the data writer that sends replies -->
-            <datawriter_qos>
-                <durability>
-                    <kind>VOLATILE_DURABILITY_QOS</kind>
-                </durability>
-            </datawriter_qos>
+        <qos_profile name="ReplierExampleProfile" base_name="BuiltinQosLib::Pattern.RPC">
 
-            <!-- QoS for the data reader that receives requests -->
```
Comment (jmuros-rti) [inline examples/connext_dds/remote_procedure_call/c++11/USER_QOS_PROFILES.xml:142]: Why are you removing the text but keep the comment? 
I think that we should keep a comment with the meaning or remove the whole comment.

File: examples/connext_dds/remote_procedure_call/py/README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Example Code: Remote Procedure Calls
 
 :warning: **Note**: For an overview of this feature, see the
```
Comment (jmuros-rti) [inline examples/connext_dds/remote_procedure_call/py/README.md:3]: Should we mention that the Python RPC is experimental?

**Review by jmuros-rti — CHANGES_REQUESTED** [review]
See the two questions

File: examples/connext_dds/remote_procedure_call/py/README.md (hunk 2 above)
Comment (alexcamposruiz) [reply examples/connext_dds/remote_procedure_call/py/README.md:3]: We already say that in the documentation. I don't think we need to say it here.

**Review by jmuros-rti — APPROVED** [review]

**Pull request merged**